lighthouse-core/audits/unused-css-rules.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit');
+const Formatter = require('../formatters/formatter');
+
+const PREVIEW_LENGTH = 100;
+const ALLOWABLE_UNUSED_RULES_RATIO = 0.10;
+
+class UnusedCSSRules extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Best Practices',
+      name: 'unused-css-rules',
+      description: 'Site does not have more than 10% unused CSS',
+      helpText: 'Remove unused rules from stylesheets to reduce unnecessary ' +
+          'bytes consumed by network activity.',
+      requiredArtifacts: ['CSSUsage', 'Styles']
+    };
+  }
+
+  static indexStylesheetsById(styles) {
+    return styles.reduce((indexed, stylesheet) => {
+      indexed[stylesheet.header.styleSheetId] = Object.assign({
+        used: [],
+        unused: [],
+      }, stylesheet);
+      return indexed;
+    }, {});
+  }
+
+  static countRules(rules, indexedStylesheets) {
+    let unused = 0;
+
+    rules.forEach(rule => {
+      const stylesheetInfo = indexedStylesheets[rule.styleSheetId];
+
+      if (rule.used) {
+        stylesheetInfo.used.push(rule);
+      } else {
+        unused++;
+        stylesheetInfo.unused.push(rule);
+      }
+    });
+
+    return {unused: unused, total: rules.length};
+  }
+
+  static mapSheetToResult(stylesheetInfo) {
+    const numUsed = stylesheetInfo.used.length;
+    const numUnused = stylesheetInfo.unused.length;
+    const percentUsed = Math.round(100 * numUsed / (numUsed + numUnused)) || 0;
+
+    let contentPreview = stylesheetInfo.content;
+    if (contentPreview.length > PREVIEW_LENGTH) {
+      const firstRuleStart = contentPreview.indexOf('{');
+      const firstRuleEnd = contentPreview.indexOf('}');
+      if (firstRuleStart === -1 || firstRuleEnd === -1
+          || firstRuleStart > firstRuleEnd
+          || firstRuleStart > PREVIEW_LENGTH) {
+        contentPreview = contentPreview.slice(0, PREVIEW_LENGTH) + '...';
+      } else if (firstRuleEnd < PREVIEW_LENGTH) {
+        contentPreview = contentPreview.slice(0, firstRuleEnd + 1) + ' ...';
+      } else {
+        const lastSemicolonIndex = contentPreview
```

Comment:
ah our line length is 100 isn't it? wish we could put that in [editorconfig](https://github.com/editorconfig/editorconfig/wiki/Property-research:-maximum-line-length)...